core/src/main/java/org/kiwix/kiwixmobile/core/downloader/ChunkUtils.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.downloader
+
+import org.kiwix.kiwixmobile.core.utils.StorageUtils.getFileNameFromUrl
+import java.util.ArrayList
+import java.util.Locale
+
+object ChunkUtils {
+  private const val ALPHABET = "abcdefghijklmnopqrstuvwxyz"
+  private const val ZIM_EXTENSION = ".zim"
+  const val PART = ".part.part"
+  const val CHUNK_SIZE = 1024L * 1024L * 1024L * 2L
+
+  fun getChunks(
+    url: String?,
+    contentLength: Long,
+    notificationID: Int
+  ): List<Chunk> {
+    val fileCount = getZimChunkFileCount(contentLength)
+    val filename = getFileNameFromUrl(url)
+    val fileNames = getZimChunkFileNames(filename, fileCount)
+    return generateChunks(contentLength, url, fileNames, notificationID)
+  }
+
+  private fun generateChunks(
+    contentLength: Long,
+    url: String?,
+    fileNames: Array<String?>,
+    notificationID: Int
+  ): List<Chunk> {
+    val chunks: MutableList<Chunk> = ArrayList()
+    var currentRange: Long = 0
+    var minLen: Long
```

Comment:
does this need to be a var?

Reply:
Since, minLen is used inside for loop, it is var

Reply:
is it used outside the loop? Is its next value dependent on the previous value? this doesn't even need to be a variable

Reply:
What I meant is similar to variable currentRange, minLen is created outside the loop as in every loop its value will change so instead of creating a new variable in every loop, I have declared it outside and used it inside.

Reply:
okay so what you have done is in simplified terms

```
var example:Long
for(i in 0.1000){
  example = i + 5
  list.add(example)
}
```
when it could be

```
for(i in 0.1000){
  list.add(i + 5)
}
```

The 2nd example is clearer because declaring a variable outside the loop implies that you need some sort of value from a previous iteration of the loop. That is why currentRange is a variable.

There is no performance gain from setting an already declared variable to a new value versus declaring a new variable, especially not one that the compiler would turn into a primitive

Reply:
@sonusourav you resolved this without implementing the change

Reply:
Hey, I am sorry, I don't know but the last commit did not get pushed somehow yesterday so that's why it was not reflecting. Now its pushed.